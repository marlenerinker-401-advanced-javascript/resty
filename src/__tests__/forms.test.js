import React from 'react';

import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Form from '../form.js';
const testFunction = jest.fn();

describe('Form tests', () => {
  it('should render at application start', () => {
    let app = shallow(<Form handler={testFunction} method={'test'} url={'test'}/>);
    expect(app.find('form').exists()).toBe(true);
    expect(app.find('input').exists()).toBe(true);
  });

  it('should update state when click GET', () => {
    let app = mount(<Form handler={testFunction} method={'test'} url={'test'}/>);
    let button = app.find({id: 'getButton', value: 'GET'});
    expect(button.exists()).toBeTruthy();
    expect(app.state('method')).toBe('GET');
    button.simulate('click');
    expect(app.state('method')).toBe('GET');
  });

  it('should update state when click POST', () => {
    let app = mount(<Form handler={testFunction} method={'test'} url={'test'}/>);
    let button = app.find({id: 'postButton', value: 'POST'});
    expect(button.exists()).toBeTruthy();
    expect(app.state('method')).toBe('GET');
    button.simulate('click');
    expect(app.state('method')).toBe('POST');
  });

  it('should update state when click PUT', () => {
    let app = mount(<Form handler={testFunction} method={'test'} url={'test'}/>);
    let button = app.find({id: 'putButton', value: 'PUT'});
    expect(button.exists()).toBeTruthy();
    expect(app.state('method')).toBe('GET');
    button.simulate('click');
    expect(app.state('method')).toBe('PUT');
  });

  it('should update state when click DELETE', () => {
    let app = mount(<Form handler={testFunction} method={'test'} url={'test'}/>);
    let button = app.find({id: 'deleteButton', value: 'DELETE'});
    expect(button.exists()).toBeTruthy();
    expect(app.state('method')).toBe('GET');
    button.simulate('click');
    expect(app.state('method')).toBe('DELETE');
  });

  it('should update state when URL is entered', () => {
    let app = mount(<Form handler={testFunction} method={'test'} url={'test'}/>);
    let input = app.find('#url');
    input.simulate('change', { target: {value: 'test.com'}});
    expect(app.state('url')).toBe('test.com');
  });


  it('renders correctly', () => {
    const DOM = renderer.create(<Form />).toJSON();
    expect(DOM).toMatchSnapshot();
  });

});
