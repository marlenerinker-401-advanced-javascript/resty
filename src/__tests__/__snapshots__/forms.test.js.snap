// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form tests renders correctly 1`] = `
<div
  id="form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      URL:
    </label>
    <input
      id="url"
      name="url"
      onChange={[Function]}
      placeholder="http://api.url.here"
      type="text"
    />
    <input
      className="action"
      type="submit"
      value="GO!"
    />
    <br />
    <input
      className="button"
      onClick={[Function]}
      type="button"
      value="GET"
    />
    <input
      className="button"
      onClick={[Function]}
      type="button"
      value="POST"
    />
    <input
      className="button"
      onClick={[Function]}
      type="button"
      value="PUT"
    />
    <input
      className="button"
      onClick={[Function]}
      type="button"
      value="DELETE"
    />
  </form>
  <section>
    <p>
      
    </p>
  </section>
</div>
`;
